### DISCUSSION TITLE
Authorization required, but no authorization protocol specified

### DISCUSSION THREAD

**Discussion by 42willow** [hyprwm/Hyprland#4222] [Q&A] [answered]
I've been having this issue ever since I installed Hyprland, meaning I can't use YaST. It's not a major problem since I don't need YaST and it has a TUI, but the GUI is a useful tool and one of the perks of using Tumbleweed.

https://forums.opensuse.org/t/authorization-required-but-no-authorization-protocol-specified/167149

### GNOME
```
~ 
❯ env | grep XAUTH
XAUTHORITY=/run/user/1000/.mutter-Xwaylandauth.3M6FG2
XAUTHLOCALHOSTNAME=localhost.localdomain
```

### Hyprland
```
~ took 2m18s 
❯ env | grep XAUTH
XAUTHLOCALHOSTNAME=localhost.localdomain
```
```
❯ xauth list
xauth:  file /home/willow/.Xauthority does not exist
```

## My configuration

https://github.com/42Willow/dotfiles/tree/main/hypr

Here is the relevant part of the autostart file:
```bash
# other
/usr/libexec/polkit-gnome-authentication-agent-1 &
dbus-update-activation-environment --systemd WAYLAND_DISPLAY XDG_CURRENT_DESKTOP &
```

## Documentation

As there is no documentation on this issue, if someone knows of a fix I think it should be added to the [environment variables](https://wiki.hyprland.org/Configuring/Environment-variables/) section of the wiki

Any help would be appreciated! Thank you!

**Comment by 42willow** [comment, marked as answer]
Edit: didn't know what I was talking about, and I'm pretty sure you don't need `xhost +local:` in your autostart-


-------------


`xhost +local: &` Fixes the issue in a way, but the ugly looking window still appears.

The environment variable and xauthority file are still missing though.

**Reply by vaxerski** [reply]
why should hyprland, a _wayland_ compositor export XAUTHORITY?

Are you prehaps missing an authentication agent? https://wiki.hyprland.org/Useful-Utilities/Must-have/#authentication-agent

**Reply by 42willow** [reply]
I have `polkit-gnome` installed and running, YaST is really the only app that has this issue, `gparted` etc all run as super user fine. Here's a related thread: https://forums.opensuse.org/t/authorization-required-but-no-authorization-protocol-specified/167149, I have tried all suggested solutions here.

Thanks for the compositor btw :)

I'll just stick with `xhost +local: &` in my autostart, it seems like a bit of a crude solution but it works.

I don't even know if `XAUTHORITY` is the issue, but there should be some kind of documentation on how to fix this issue *somewhere*, maybe is should open an issue in the opensuse issue tracker (wherever that is) because this occurs on sway too (or so I've heard)

**Reply by blacklightpy** [reply]
`xhost +SI:local:` is a more secure choice, as otherwise any user from any host named local to connect to your machine, as opposed to setting the family to `SI` (Server Identified).

**Reply by 42willow** [reply]
Thanks, I'll use that instead

**Reply by blacklightpy** [reply]
> Thanks, I'll use that instead

Sorry, `+local:` seems like the better choice if you are adding all users in the local network. I was using `+SI:localuser:root` so I thought `+SI:localuser:` would be the right choice for setting all users. 

`+SI:local:` won't probably mean anything as that was a mistake.

> A complete name has the syntax ‘‘family:name’’ where the families are as follows:
> 
> inet        Internet host (IPv4)
> inet6      Internet host (IPv6)
> dnet       DECnet host
> nis          Secure RPC network name
> krb         Kerberos V5 principal
> local       contains only one name, the empty string
> si             Server Interpreted

> The local family specifies all the local connections at once. However, the server interpreted address "si:localuser:username" can be used to specify a single local user. (See the Xsecurity(7) manual page for more details.)

**Comment by vuongtuha** [comment]
Excuse me, I don't understand well how did you fix your issue. Not only Gparted, but many Gnome app got this issue on Hyprland. Hope you can tell me more clear what config did you added. Thanks!

**Reply by fufexan** [reply]
You need an authentication agent for Polkit. See https://wiki.archlinux.org/title/Polkit#Authentication_agents.

**Reply by vuongtuha** [reply]
then I put this app ( in case Gnome, it's called polkit-gnome-authentication-agent-1)  to autostart config on hypr config. 
And put more "xhost +local: &" to autostart. Is that right?

**Reply by fufexan** [reply]
You don't need xhost if you have the auth agent working. And yes, you put the gnome agent in exec-once.

**Reply by blacklightpy** [reply]
@fufexan I had kde agent and this was still a problem..

**Reply by fufexan** [reply]
Works just fine for me with the gnome agent. Never used xhost, and only in rare cases do I need to use `sudo -E`.

**Reply by 42willow** [reply]
```bash
#!/usr/bin/bash

# cursor theme
hyprctl setcursor Catppuccin-Macchiato-Dark-Cursors 24 &

# bar
waybar &

# update packages list
pacman -Qe | awk '{print $1}' > $HOME/.dotfiles/assets/packages.txt &

# clipboard
wl-paste --type text --watch cliphist store & #Stores only text data

# wallpaper
hyprpaper &

# other
/usr/lib/polkit-gnome/polkit-gnome-authentication-agent-1 &
dbus-update-activation-environment --systemd WAYLAND_DISPLAY XDG_CURRENT_DESKTOP &
```

This is my autostart, try `xhost +local:` if you're having issues. I 98% sure you don't need the xhost thing in autostart... I didn't know what I was talking about.

I don't have this issue anymore on Arch, only happened on tumbleweed.

Here's the output of `xhost` for me:
```
❯ xhost
access control enabled, only authorized clients can connect
SI:localuser:willow
```

**Reply by blacklightpy** [reply]
@fufexan I think it's got something to do with GTK apps.. For me, `gnome-control-center` used to need `env XDG_CURRENT_DESKTOP=GNOME` set to run.

**Reply by fufexan** [reply]
Yes, but that's only the control center. The other programs should work just fine.
